### PR TITLE
feat: use buildpack

### PR DESCRIPTION
**Pull request by viceice**
ref: #15

### CODE REVIEW

**Comment by viceice** [comment]
```
DEBUG: Executing command (repository=renovate-tests/poetry1, branch=renovate-test/pytest-5.x)
       "command": "poetry update --lock --no-interaction pytest"
DEBUG: Failed to update poetry.lock file (repository=renovate-tests/poetry1, branch=renovate-test/pytest-5.x)
       "err": {
         "killed": false,
         "code": 126,
         "signal": null,
         "cmd": "poetry update --lock --no-interaction pytest",
         "stdout": "",
         "stderr": "/bin/sh: 1: poetry: Permission denied\n",
         "message": "Command failed: poetry update --lock --no-interaction pytest\n/bin/sh: 1: poetry: Permission denied\n",
         "stack": "Error: Command failed: poetry update --lock --no-interaction pytest\n/bin/sh: 1: poetry: Permission denied\n\n    at ChildProcess.exithandler (child_process.js:303:12)\n    at ChildProcess.e
mit (events.js:310:20)\n    at ChildProcess.EventEmitter.emit (domain.js:482:12)\n    at maybeClose (internal/child_process.js:1021:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:28
6:5)"
       }
```

File: Dockerfile [hunk 1]
```diff
@@ -2,32 +2,53 @@ ARG IMAGE=latest
...
+  org.opencontainers.image.licenses="AGPL-3.0-only"
 
 USER root
 WORKDIR /usr/src/app/
 
+# Python
+
+# renovate: datasource=docker depName=python
+ENV PYTHON_VERSION=3.8.2
+RUN install-tool python
```
Comment (rarkins) [inline Dockerfile:20]:
```suggestion
RUN install-tool python 3.8.2
```

Is this possible now?

Comment (viceice) [reply Dockerfile:20]: sure, should we do this for all tools?

File: Dockerfile [hunk 2]
```diff
@@ -64,149 +74,82 @@ ENV RENOVATE_BINARY_SOURCE=docker
...
-# renovate: datasource=npm depName=npm versioning=npm
-ARG PNPM_VERSION=4.12.0
-RUN /usr/local/build/pnpm.sh
-
-USER ubuntu
 
-# HOME does not get passed after user switch :-(
-ENV HOME=/home/ubuntu
+# renovate: datasource=docker versioning=docker
+RUN install-tool ruby 2.5.8
```
Comment (rarkins) [inline Dockerfile:135]: Is there a problem with running cocoapods with latest Ruby?

File: Dockerfile [hunk 3]
```diff
@@ -64,149 +74,82 @@ ENV RENOVATE_BINARY_SOURCE=docker
...
+# renovate: datasource=npm versioning=npm
+RUN install-tool pnpm 4.12.0
 
-RUN set -ex ;\
-  curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path --profile minimal --default-toolchain ${RUST_VERSION} -y
+USER ubuntu
 
 # Mix and Rebar
 
 RUN mix local.hex --force
```
Comment (rarkins) [inline Dockerfile:149]: Can this be done using `install-tool` scripts?

File: Dockerfile (hunk 2 above)
Comment (viceice) [reply Dockerfile:135]: not yet tested, want to make sure everything works as before, so renovate can propose the update

File: Dockerfile [hunk 4]
```diff
@@ -218,8 +161,16 @@ COPY --from=tsbuild /usr/src/app/dist dist
...
-ENTRYPOINT ["node", "/usr/src/app/dist/renovate.js"]
-CMD []
+
+COPY bin/ /usr/local/bin/
+
+RUN set -ex; \
+  ln -sf /usr/src/app/dist/renovate.js /usr/local/bin/renovate; \
+  renovate --version;
+
+ENTRYPOINT ["docker-entrypoint.sh"]
```
Comment (rarkins) [inline Dockerfile:171]: This isn't inherited from buildpack?

File: Dockerfile (hunk 3 above)
Comment (viceice) [reply Dockerfile:149]: mayke we can install it globally too? I'll check this

File: bin/docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+. /usr/local/docker/env
+
+if [[ "${1:0:1}" = '-' ]]; then
+  # assume $1 is renovate flag
```
Comment (rarkins) [inline bin/docker-entrypoint.sh:6]: Is this fallback new?

File: Dockerfile (hunk 4 above)
Comment (viceice) [reply Dockerfile:171]: nope, I've added some sugar to use the image more easily

File: bin/docker-entrypoint.sh (hunk 5 above)
Comment (viceice) [reply bin/docker-entrypoint.sh:6]: yes, so you can now run `docker run --rm -it renovate/renovate bash` useful for debugging

File: Dockerfile (hunk 4 above)
Comment (rarkins) [reply Dockerfile]: Documented?

File: Dockerfile [hunk 6]
```diff
@@ -2,32 +2,50 @@ ARG IMAGE=latest
...
 
 LABEL maintainer="Rhys Arkins <rhys@arkins.net>"
 LABEL name="renovate"
 LABEL org.opencontainers.image.source="https://github.com/renovatebot/renovate" \
-      org.opencontainers.image.url="https://renovatebot.com" \
-      org.opencontainers.image.licenses="AGPL-3.0-only"
+  org.opencontainers.image.url="https://renovatebot.com" \
+  org.opencontainers.image.licenses="AGPL-3.0-only"
 
 USER root
```
Comment (rarkins) [inline Dockerfile:13]:
```suggestion
```

buildpack defaults to ROOT

File: Dockerfile (hunk 4 above)
Comment (viceice) [reply Dockerfile]: A little bit, see readme changes. I'll add samples for gitlab and jenkins

File: Dockerfile [hunk 7]
```diff
@@ -74,71 +65,48 @@ ENV RENOVATE_BINARY_SOURCE=docker
 #============
 FROM final-base as latest
 
-# General tools
-
 # go suggests: git svn bzr mercurial
 RUN install-apt bzr mercurial
```
Comment (rarkins) [inline Dockerfile:69]: Is there anything preventing us from installing this as part of `install-tool golang`?

Comment (viceice) [reply Dockerfile]: Yes, they are not strictly dependencies for go. They would only needed when using sources from those repository types

**Pull request merged**